app/src/main/java/com/kamron/pogoiv/Pokefly.java

```diff
@@ -695,6 +698,37 @@ private void createInfoLayout() {
         createInputLayout();
         createResultLayout();
         createAllIvLayout();
+
+        initPositionHandler();
+    }
+
+    private void initPositionHandler() {
+        positionHandler.setOnTouchListener(new View.OnTouchListener() {
+            WindowManager.LayoutParams newParams = layoutParams;
+            double y, startingY;
+
+            @Override
+            public boolean onTouch(View v, MotionEvent event) {
+                switch (event.getAction()) {
+                    case MotionEvent.ACTION_DOWN:
+                        y = newParams.y;
+                        startingY = event.getRawY();
+                        break;
+
+                    case MotionEvent.ACTION_MOVE:
+                        if(appraisalBox.getVisibility()==View.VISIBLE){
+                            newParams.y = (int) (y + (event.getRawY() - startingY));
+                        }else {
+                            newParams.y = (int) (y - (event.getRawY() - startingY));
```

Comment:
Do we need to remember `y`, rather than use `newParams.y` directly? Or maybe that's a silly question...